memory/transparent_hugepages_swapping.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Test Inspired basically from
+# https://github.com/autotest/tp-qemu/blob/master/generic/tests/trans_hugepage_swapping.py
+#
+# Copyright: 2017 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.core import data_dir
+from avocado.utils.partition import Partition
+
+
+class Thp_Swapping(Test):
+
+    '''
+    The test fills out the total avl memory and tries to swap the thp out.
+    '''
+
+    def setUp(self):
+
+        '''
+        Sets the Required params for dd and mounts the tmpfs dir
+        '''
+
+        self.swap_free = []
+        mem_free = memory.read_from_meminfo("MemFree") / 1024
+        mem = memory.read_from_meminfo("MemTotal") / 1024
+        tmpdir = data_dir.get_tmp_dir()
+        swap = memory.read_from_meminfo("SwapTotal") / 1024
+        hugepage_size = memory.read_from_meminfo("Hugepagesize") / 1024
+        self.swap_free.append(memory.read_from_meminfo("SwapFree") / 1024)
+        # Set Block size as hugepage size * 2
+        self.block_size = (memory.read_from_meminfo("Hugepagesize") / 1024) * 2
```

Comment:
or better where you want to use self.block_size there just use `hugepage_size/2`